cypress/e2e/signIn.cy.js

```diff
@@ -12,23 +17,54 @@ describe('Sign In page', () => {
 
   before(() => {
     cy.task('db:clear');
+  });
+
+  beforeEach(() => {
     cy.task('generateUser').then((generateUser) => {
       user = generateUser;
+      signInPage.visit();
+      cy.registerUser(user.email, user.username, user.password);
     });
   });
 
   it('should provide an ability to log in with existing credentials', () => {
-    signInPage.visit();
-    cy.register(user.email, user.username, user.password);
-
-    signInPage.typeEmail(user.email);
-    signInPage.typePassword(user.password);
-    signInPage.clickSignInBtn();
+    signInPage.enterLoginCredentials(user.email, user.password);
+    signInPage.clickOnSignInButton();
 
     homePage.assertHeaderContainUsername(user.username);
```

Comment:
The `assertHeaderContainUsername` method is not defined in the HomePageObject class. Make sure you have it properly defined and imported.

Reply:
![e2e_16](https://github.com/mate-academy/qa_cypress_e2e_project/assets/13405550/0f5efbe6-1c5c-4e62-9512-fb6f02440424)